pkg/profile/artifact.go

```diff
@@ -53,68 +56,65 @@ func (p *Profile) CreateArtifacts(ctx context.Context) error {
 
 // ArtifactStatus checks if the artifacts exists and returns any ready!=true conditions on the artifacts.
 func (p *Profile) ArtifactStatus(ctx context.Context) (Status, error) {
-	resourcesExist, gitRes, helmRes, kustomizeRes, err := p.getResources(ctx)
+	resourcesExist, gitRes, helmResources, kustomizeResources, err := p.getResources(ctx)
 	if err != nil {
 		return Status{}, err
 	}
 
 	return Status{
 		ResourcesExist:     resourcesExist,
-		NotReadyConditions: p.checkResourcesReady(gitRes, helmRes, kustomizeRes),
+		NotReadyConditions: p.checkResourcesReady(gitRes, helmResources, kustomizeResources),
 	}, nil
 }
 
-func (p *Profile) checkResourcesReady(gitRes *sourcev1.GitRepository, helmRes *helmv2.HelmRelease, kustomizeRes *kustomizev1.Kustomization) []metav1.Condition {
+func (p *Profile) checkResourcesReady(gitRes *sourcev1.GitRepository, helmResources []*helmv2.HelmRelease, kustomizeResources []*kustomizev1.Kustomization) []metav1.Condition {
 	var notReadyConditions []metav1.Condition
 	if gitRes != nil {
 		if condition := getNotReadyCondition(gitRes.Status.Conditions); condition != (metav1.Condition{}) {
 			notReadyConditions = append(notReadyConditions, condition)
 		}
 	}
 
-	if helmRes != nil {
+	for _, helmRes := range helmResources {
 		if condition := getNotReadyCondition(helmRes.Status.Conditions); condition != (metav1.Condition{}) {
 			notReadyConditions = append(notReadyConditions, condition)
 		}
 	}
 
-	if kustomizeRes != nil {
+	for _, kustomizeRes := range kustomizeResources {
 		if condition := getNotReadyCondition(kustomizeRes.Status.Conditions); condition != (metav1.Condition{}) {
 			notReadyConditions = append(notReadyConditions, condition)
 		}
 	}
 	return notReadyConditions
 }
 
-func (p *Profile) getResources(ctx context.Context) (bool, *sourcev1.GitRepository, *helmv2.HelmRelease, *kustomizev1.Kustomization, error) {
-	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+func (p *Profile) getResources(ctx context.Context) (bool, *sourcev1.GitRepository, []*helmv2.HelmRelease, []*kustomizev1.Kustomization, error) {
```

Comment:
Question:
Can't there be multiple GitRepositories with possible multiple paths which are all copied into the same profile repository?

Reply:
Atm we only use 1 `GitRepostory` resource as it only ever points to the profile definition repo, e.g.:
```
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: nginx-profile-test-nginx-profile-multiple-artifacts
  namespace: default
spec:
  gitImplementation: go-git
  interval: 0s
  ref:
    branch: multiple-artifacts
  timeout: 20s
  url: https://github.com/weaveworks/nginx-profile
```

So we only need 1 of them. As part of https://github.com/weaveworks/profiles/pull/54 theres actually now a scenario where you don't need this resource, if the profile only contains remote helm chart references.